gdal/swig/include/csharp/typemaps_csharp.i

```diff
@@ -33,6 +33,45 @@
 
 /* CSHARP TYPEMAPS */
 
+%typemap(csdispose) SWIGTYPE %{
```

Comment:
shouldn't those be only defined for SWIG < 4 (at least for clarity) ?

Reply:
No, I think they are needed for SWIG >= 4. It's providing some kind of backward compat. It should be possible to do changes so that it should not be needed but I don't understand SWIG good enough yet.

Reply:
Nobody understands SWIG, the simple fact of daring to touch SWIG related stuff makes you a fully qualified SWIG developer, so you're on the right track ;-)

I guess I'm confused because https://github.com/swig/swig/commit/06462acdf9b7738a19dce5579d90b1c91b3f9cfb#diff-8356d5285b5b342a204d4ff54fc34973 seems to introduce typemaps with the same names as the one you added, but slightly different definitions

Reply:
Right.. :) I've got it fully working locally now at least. Perhaps, in time, I can reach some further conclusions on what is the most appropriate changes.